client/app/dedicated/server/firewall/dedicated-server-firewall.controller.js

```diff
@@ -1,44 +1,77 @@
-angular.module("App").controller("ServerTabFirewallAsaCtrl", ($scope, $q, $stateParams, Server, ServerFirewallAsa) => {
-    $scope.informations = null;
-    $scope.server = null;
-    $scope.canOrderAsaOption = false;
-    $scope.loading = true;
+angular.module("App")
+    .controller("ServerTabFirewallAsaCtrl", class ServerTabFirewallAsaCtrl {
+        constructor ($q, $scope, $stateParams, Alerter, Server, ServerFirewallAsa, translator) {
+            this.$q = $q;
+            this.$scope = $scope;
+            this.$stateParams = $stateParams;
+            this.Alerter = Alerter;
+            this.Server = Server;
+            this.ServerFirewallAsa = ServerFirewallAsa;
+            this.translator = translator;
 
-    function init () {
-        $scope.informations = null;
-        $q.allSettled([initServer(), initOptions(), initInformations()]).finally(() => {
-            $scope.loading = false;
-        });
-    }
+            this.$scope.$on(this.ServerFirewallAsa.events.firewallAsaChanged, () => this.$onInit());
+        }
 
-    function initServer () {
-        return Server.getSelected($stateParams.productId).then((server) => {
-            $scope.server = server;
-        });
-    }
+        $onInit () {
+            this.server = null;
+            this.canOrderAsaOption = false;
+            this.informations = null;
 
-    function initOptions () {
-        return ServerFirewallAsa.getOptionList($stateParams.productId).then((availableOptions) => {
-            if (availableOptions && availableOptions.results.length > 0) {
-                $scope.canOrderAsaOption = true;
-            }
-        });
-    }
+            this.isLoading = true;
+            return this.$q
+                .all({
+                    server: this.getServer(),
+                    optionList: this.getOptionList(),
+                    informations: this.getInformations()
+                })
+                .finally(() => {
```

Comment:
Maybe you shoud catch and display an error here and reject in others methods ... but catch the potential error :D